keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplier.kt

```diff
@@ -101,10 +101,31 @@ class DebianArtifactSupplier(
     return null
   }
 
-  // Debian Artifacts should contain a releaseStatus in the metadata
-  private fun PublishedArtifact.hasReleaseStatus() =
-    this.metadata.containsKey("releaseStatus") && this.metadata["releaseStatus"] != null
 
   private val DeliveryArtifact.statusesForQuery: List<String>
     get() = statuses.map { it.name }
+
+  override fun shouldProcessArtifact(artifact: PublishedArtifact): Boolean =
+    artifact.hasReleaseStatus() && artifact.hasCorrectVersion()
+
+
+  // Debian Artifacts should contain a releaseStatus in the metadata	  // Debian Artifacts should contain a releaseStatus in the metadata
+  private fun PublishedArtifact.hasReleaseStatus() : Boolean {
+    return if (this.metadata.containsKey("releaseStatus") && this.metadata["releaseStatus"] != null) {
+      true
+    } else {
+      log.debug("Ignoring artifact event without release status: $this")
+      false
+    }
+  }
+
+  // Debian Artifacts should not have "local" as a part of their version string
+  private fun PublishedArtifact.hasCorrectVersion() : Boolean {
+    return if (!this.version.contains("local")) {
```

Comment:
I think we only care about "local" in a specific spot. I think it would be better to parse with frigga here and then get the build number to check if that is == local

Reply:
ohhhh good point!!!!